### DISCUSSION TITLE
Notes about porting to psycopg 3

### DISCUSSION THREAD

**Discussion by dvarrazzo** [dbcli/pgcli#1317] [Ideas]
Just dropping a few comments here, after the last messages in #1085, and have a conversation about porting pgcli to pyscopg3.

I've taken a look at pgcli source and the way it interacts with psycopg 2: there will be probably no big problem in porting.

One thing I notice is that you use a green callback so that ctrl-c can interrupt the query. Psycopg 3, in a way, always works in green mode. However there is currently no provision for changing the callback. At the moment Ctrl-C does stop the query, but breaks the connection, leaving in with an open command in progress. This is something that should be probably improved in psycopg 3: https://github.com/psycopg/psycopg/issues/231

**Comment by dbaty** [comment]
pgcli has been ported to psycopg 3 since your message. I'll hence close this discussion.

Thanks again for chiming in and your support here and in the aforementioned pull request (and obviously, in your great work on psycopg), this is much appreciated. :)

**Reply by dvarrazzo** [reply]
Thank you very much for a very pleasant collaboration and for finding so many bugs in psycopg! I hope it is now serving you well! :)

**Reply by j-bennet** [reply]
We love psycopg. :)
